src/safeds/data/tabular/transformation/_one_hot_encoder.py

```diff
@@ -119,37 +134,49 @@ def transform(self, table: Table) -> Table:
             If the transformer has not been fitted yet.
         """
         # Transformer has not been fitted yet
-        if self._wrapped_transformer is None or self._column_names is None:
+        if self._column_names is None or self._value_to_column is None:
             raise TransformerNotFittedError
 
         # Input table does not contain all columns used to fit the transformer
         missing_columns = set(self._column_names.keys()) - set(table.column_names)
         if len(missing_columns) > 0:
             raise UnknownColumnNameError(list(missing_columns))
 
-        original = table._data.copy()
-        original.columns = table.schema.column_names
+        # Make a copy of the table:
+        # TODO: change to copy method once implemented
+        new_table = table.remove_columns([])
```

Comment:
Is the copy really necessary? At first glance, none of the operations you call on `new_table` mutate its internal state.

Reply:
You're right, the copy is not neccessary. Removed it.

We do, however, need to make sure the column count can't temporarily drop to 0 (that would confuse the `table` class), so I moved the part where we remove the old columns to the bottom.